src/mpl2/src/object.cpp

```diff
@@ -973,14 +977,22 @@ void HardMacro::updateDb(float pitch_x, float pitch_y)
   ux = std::round(ux / pitch_x) * pitch_x;
   ly = std::round(ly / pitch_y) * pitch_y;
   uy = std::round(uy / pitch_y) * pitch_y;
+  int round_lx = std::round(micronToDbu(lx, dbu_) / manufacturing_grid_)
+                 * manufacturing_grid_;
```

Comment:
You are doing integer division so there will be nothing to round.  I expect you meant to do floating point division.

Reply:
lx is a float in user units

This is another change that we need to tackle to convert  all the internal val;ues to dbu (int) instead of user units.

Reply:
you are right -- I converted the dividend to float -- the divide and multiply by manufacturing grid is still needed to snap the loc to closest manufacturing grid